src/Psalm/Internal/Type/SimpleAssertionReconciler.php

```diff
@@ -2507,4 +2517,50 @@ function (Type\Union $t): bool {
             ? Type::getEmpty()
             : Type::getMixed();
     }
+
+    /**
+     * @param   0|1|2    $failed_reconciliation
+     */
+    private static function reconcileClassConstant(
+        Codebase $codebase,
+        string $class_constant_expression,
+        int &$failed_reconciliation
+    ) : Union {
+        if (strpos($class_constant_expression, '::') === false) {
+            $failed_reconciliation = 2;
+            return Type::getMixed();
+        }
+
+        [$class_name, $constant_pattern] = explode('::', $class_constant_expression, 2);
+
+        if (!$codebase->classlike_storage_provider->has($class_name)) {
+            $failed_reconciliation = 2;
+            return Type::getMixed();
+        }
+
+        $class_like_storage = $codebase->classlike_storage_provider->get($class_name);
+        $matched_class_constant_types = [];
+
+        foreach ($class_like_storage->constants as $constant => $class_constant_storage) {
+            if (!fnmatch($constant_pattern, $constant)) {
```

Comment:
I'm a little uneasy using a function made for filenames in this context. I think we should already have a similar feature elsewhere without using that

Reply:
other parts remove the wildcard and then use `strpos`.
Should I do so as well?

```php
$const_name_part = \substr($return_type->const_name, 0, -1);

if ($const_name_part) {
    $matching_constants = \array_filter(
        $matching_constants,
        function ($constant_name) use ($const_name_part): bool {
            return $constant_name !== $const_name_part
                && \strpos($constant_name, $const_name_part) === 0;
        }
    );
}
```

Reply:
that doesn't seem a lot better to me. Maybe a small function replacing * with .* in a regex would be better?

Reply:
done